internal/controller/helmchart_controller.go

```diff
@@ -884,6 +888,7 @@ func (r *HelmChartReconciler) reconcileArtifact(ctx context.Context, _ *patch.Se
 	// Record it on the object
 	obj.Status.Artifact = artifact.DeepCopy()
 	obj.Status.ObservedChartName = b.Name
+	obj.Status.ObservedValuesFiles = b.ValuesFiles
```

Comment:
I have been thinking about this a bit more. I think we should do this only if the new `IgnoreMissingValuesFiles` is set. Mainly because existing `HC custom resources` with cached charts won't have this logic and it feels wrong to have an empty `obj.Status.ObservedValuesFiles` where it should be populated. Setting up `IgnoreMissingValuesFiles` for an existing custom resource with cached charts should report all the valuesFiles because otherwise it should have errored in previous reconciliation.

Reply:
Updated.